modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/UniqueTokenFilter.java

```diff
@@ -74,6 +81,8 @@ public final boolean incrementToken() throws IOException {
             if (!duplicate) {
                 return true;
             }
+
+            skippedIncrement = posIncrement;
```

Comment:
This isn't quite right, because we need to take into account multiple skipped tokens.